lib/utils/utils.dart

```diff
@@ -452,8 +455,48 @@ class UrlUtils {
     }
     return url;
   }
+
+  static Future<bool> canLaunchUrl(String url) async {
+    Uri? parsedUri = Uri.tryParse(url);
+    return (parsedUri != null) && (await url_launcher.canLaunchUrl(parsedUri));
+  }
+
+  ///
+  /// Launches urls based on the mode. 'context' must be specified if the mode is UrlLaunchMode.internalWebPanel
+  ///
+  /// returns true if the url is launched and false otherwise
+  ///
+  static void launchUrl({required String url, BuildContext? context, UrlLaunchMode? mode}) {
+    switch (mode) {
```

Comment:
Should we call `canLaunchUrl` on `url` at the beginning here?

Reply:
Yes, you are right @shurwit , I also think we should call it.